### DISCUSSION TITLE
How to identify if a realtime connection was closed due to inactivity?

### DISCUSSION THREAD

**Discussion by erickythierry** [pocketbase/pocketbase#4916] [Q&A]
greetings everyone.
I created a project that uses the realtime db function to identify changes in data and perform processing (indirect communication between the front and back through the db).
However, I noticed that after a few hours without changes occurring, the realtime connection with the db stops and the backend does not identify new changes if they occur.

Is there any way to identify if the realtime listener has terminated so that it can be restarted?

**Comment by ganigeorgiev** [comment]
It is not clear what is the issue, nor what you are asking.

If you are using the SDKs the client will try to reconnect in case the connection was interrupted (eg. on idle timeout) and users don't need to do anything in addition.

**Reply by erickythierry** [reply]
Thanks for the reply, sorry for my vague question, it's just that I couldn't quite understand what's going on. 
I'm using the JS SDK, and I have a backend that is just waiting for modifications in a DB collection, however, from one day to the next, I ended up noticing that the backend stopped "listening" for the modifications of the DB collection. 

I had to restart the code for it to "hear" the modifications again 

so I ended up asking if there is any way to identify if the Realtime DB listener stops listening for modifications.

I noticed that this behavior was occurring at night/early in the morning, when there was no change in the collection for long periods.

**Reply by ganigeorgiev** [reply]
There is no "disconnect" SDK event if that's what you are asking.

The realtime connection is automatically restarted after 5mins idle timeout so as long as your client app is running and you haven't unsubscribed from your topic, it should work correctly. If your use case requires to be authenticated make sure that your client token hasn't expired (_you can attach to the `pb.realtime.subscribe("PB_CONNECT", () => { ... })` and call `authRefresh` if you need to or resubscribe after manual authentication_).

Without clear reproducible steps or information about your setup to attempt to reproduce it locally I'm not sure how to help.

**Reply by ganigeorgiev** [reply]
Note that you can change the idle timeout if you think that the reconnect is breaking something in your app via the [`OnRealtimeConnectRequest` hook](https://pocketbase.io/docs/go-event-hooks/#onrealtimeconnectrequest), something like:

```go
app.OnRealtimeConnectRequest().Add(func(e *core.RealtimeConnectEvent) error {
	e.IdleTimeout = 168 * time.Hour // ~ 1 week
	return nil
})
```

**Reply by erickythierry** [reply]
Thanks for the explanation, I believe it could be the authentication so, as I don't use unsubscribe at any time, I will review this authentication part to refresh it from time to time. I believe this could be it.

**Comment by erickythierry** [comment]
> There is no "disconnect" SDK event if that's what you are asking.
> 
> The realtime connection is automatically restarted after 5mins idle timeout so as long as your client app is running and you haven't unsubscribed from your topic, it should work correctly. If your use case requires to be authenticated make sure that your client token hasn't expired (_you can attach to the `pb.realtime.subscribe("PB_CONNECT", () => { ... })` and call `authRefresh` if you need to or resubscribe after manual authentication_).
> 
> Without clear reproducible steps or information about your setup to attempt to reproduce it locally I'm not sure how to help.
